backend/pkg/backend/backend.go

```diff
@@ -306,13 +310,51 @@ func (b *Backend) handleHTTPTrace(trace *_spec.SCNTelemetry) error {
 			event.NewProvidedSpec = string(modified)
 		}
 		event.PathID = providedDiff.PathID
+		providedDiffType = getAPIDiffType(providedDiff.Type)
 	}
 
+	event.SpecDiffType = getSpecDiffType(providedDiffType, reconstructedDiffType)
```

Comment:
Why are we taking the higher specDiffType, and not creating providedspecDiffType and reconstructedspecDiffType on the ApiEvent?

Reply:
yes by design

Reply:
@fishkerez I think that the DB query will be harder to merge the type on fetch.